src/gwt/src/org/rstudio/studio/client/workbench/ui/PaneManager.java

```diff
@@ -382,6 +382,14 @@ else if (numDocs > 0 && window.getState() == WindowState.HIDE)
          hiddenTabs_ = tabNamesToTabs(evt.getValue().getHiddenTabSet());
          populateTabPanel(hiddenTabs_, hiddenTabSetTabPanel_, hiddenTabSetMinPanel_);
 
+         // manage source column commands
+         boolean visible = userPrefs.allowSourceColumns().getValue() &&
+            (userPrefs.panes().getValue().getAdditionalSourceColumns() < MAX_COLUMN_COUNT);
+         
+         commands_.newSourceColumn().setVisible(visible);
+         commands_.openSourceDocNewColumn().setVisible(visible);
+         commands_.focusSourceColumnSeparator().setVisible(visible);
```

Comment:
Here and below: it makes sense to disable the first two commands if we're already maxed out for columns (since they create columns), but it seems to me like the third shouldn't be affected. What do you think?

Reply:
You're right, fixed it.